client/app/src/main/java/group01/smartcar/client/CarControl.java

```diff
@@ -232,4 +240,64 @@ private void updateSpeedometer(){
         speedometer.setText(getCurrentSpeedKMHString());
     }
 
+    public void voiceControl(String results) {
+        ArrayList<String> dictionary = new ArrayList<String>( Arrays.asList("forward", "reverse", "stop", "speed", "turn", "left", "right", "lean", "zero", "one", "two", "three", "four", "five", "0", "1", "2", "3", "4", "5") );
+        ArrayList<String> trimmedResults = new ArrayList(Arrays.asList(results.toLowerCase().split(" ")));
+        ArrayList<String> cleanResults = new ArrayList();
```

Comment:
You should try to program to an interface instead of the implementation whenever possible. Instead of initializing your lists as:

```java
ArrayList<String> list = new ArrayList<>();
```

you can initialize it like this:

```java
List<String> list = new ArrayList<>();
```

This decouples the interface from the actual implementation.

Doing this you will also be able to properly use `Arrays#asList`. Currently, you're unnecessarily initializing two lists `new ArrayList<String>(Arrays.asList(...))` because you're locking yourself into the specific ArrayList implementation. If you switch to List, you're able to initialize your list like this:

```java
List<String> list = Arrays.asList(...);
```

Reply:
updated!